mozetl/taar/taar_similarity.py

```diff
@@ -36,26 +41,27 @@ def get_samples(spark):
     """ Get a DataFrame with a valid set of sample to base the next
     processing on.
     """
-    return (
+    df = (
         spark.sql("SELECT * FROM longitudinal")
         .where("active_addons IS NOT null")
         .where("size(active_addons[0]) > 2")
         .where("size(active_addons[0]) < 100")
         .where("normalized_channel = 'release'")
         .where("build IS NOT NULL AND build[0].application_name = 'Firefox'")
-        .selectExpr(
-            "client_id as client_id",
-            "active_addons[0] as active_addons",
-            "geo_city[0] as geo_city",
-            "subsession_length[0] as subsession_length",
-            "settings[0].locale as locale",
-            "os as os",
-            "places_bookmarks_count[0].sum AS bookmark_count",
-            "scalar_parent_browser_engagement_tab_open_event_count[0].value AS tab_open_count",
-            "scalar_parent_browser_engagement_total_uri_count[0].value AS total_uri",
-            "scalar_parent_browser_engagement_unique_domains_count[0].value AS unique_tlds"
-        )
+    ).selectExpr(
+        "client_id as client_id",
+        "active_addons[0] as active_addons",
+        "geo_city[0] as geo_city",
+        "subsession_length[0] as subsession_length",
+        "settings[0].locale as locale",
+        "os as os",
+        "places_bookmarks_count[0].sum AS bookmark_count",
+        "scalar_parent_browser_engagement_tab_open_event_count[0].value AS tab_open_count",
+        "scalar_parent_browser_engagement_total_uri_count[0].value AS total_uri",
+        "scalar_parent_browser_engagement_unique_domains_count[0].value AS unique_tlds",
     )
+    df.cache()
```

Comment:
If we're going to put this in as-is, let's add a comment explaining why this was added as a workaround.